### PR TITLE
Upgrade to Shopsys Framework version 9-dev

### PR DESCRIPTION
**Pull request by TomasLudvik**
| Q             | A
| ------------- | ---
|Description, reason for the PR| ...
|New feature| Yes 
|Fixes issues| ...
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: tests/App/Performance/Feed/AllFeedsTest.php [hunk 1]
```diff
@@ -64,7 +64,7 @@ public function testAllFeedsGeneration()
         $performanceTestSamples = [];
         $allFeedGenerationData = $this->getAllFeedGenerationData();
         foreach ($allFeedGenerationData as $feedGenerationData) {
-            list($feedInfo, $domainConfig, $maxDuration) = $feedGenerationData;
+            [$feedInfo, $domainConfig, $maxDuration] = $feedGenerationData;
```
Comment (pesektomas) [inline tests/App/Performance/Feed/AllFeedsTest.php:67]: :+1:

File: config/images.yml [hunk 2]
```diff
@@ -36,8 +39,8 @@
     class: Shopsys\FrameworkBundle\Model\Category\Category
     sizes:
         -   name: ~
-            width: 70
-            height: 70
+            width: 30
+            height: 30
```
Comment (pesektomas) [inline config/images.yml]: hmm, why did you change this values?

File: config/images.yml [hunk 3]
```diff
@@ -117,30 +120,6 @@
...
-        -   name: homepageSecondaryFirst
-            width: 439
-            height: 160
-            crop: false
-            occurrence: 'Front-end: Advertising on homepage above popular categories (Secondary First)'
-        -   name: homepageSecondarySecond
-            width: 439
-            height: 160
-            crop: false
-            occurrence: 'Front-end: Advertising on homepage above popular categories (Secondary Second)'
```
Comment (pesektomas) [inline config/images.yml]: And why did you remove this configurations? I don't know if this change will broke some shop functionality?

File: config/packages/fos_ck_editor.yaml [hunk 4]
```diff
@@ -32,8 +32,7 @@ fos_ck_editor:
                 - ['Outdent', 'Indent', '-', 'Blockquote']
                 - ['Link', 'Unlink']
                 - '/'
-                - ['Source', '-', 'Image', 'Table', 'SpecialChar']
+                - ['Source', '-', 'Image', 'SpecialChar']
```
Comment (pesektomas) [inline config/packages/fos_ck_editor.yaml]: Is OK that you removed Table functionality from CK editor?

File: config/packages/frontend_api.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+parameters:
+    shopsys.frontend_api.keys_filepath: '%shopsys.root_dir%/config/frontend-api'
+    shopsys.frontend_api.domains:
+        - 1
+        - 2
```
Comment (pesektomas) [inline config/packages/frontend_api.yaml:5]: it is good that fe api is allowed :+1:

**Review by pesektomas — COMMENTED** [review]
Hello @TomasLudvik,  good luck in your next work.

File: config/packages/fos_ck_editor.yaml (hunk 4 above)
Comment (TomasLudvik) [reply config/packages/fos_ck_editor.yaml]: Fixed in 268abd1bae86f02635c43016b0523475fff3663c

File: config/images.yml (hunk 3 above)
Comment (TomasLudvik) [reply config/images.yml]: Fixed in 268abd1bae86f02635c43016b0523475fff3663c

File: config/images.yml (hunk 2 above)
Comment (TomasLudvik) [reply config/images.yml]: Fixed in 268abd1bae86f02635c43016b0523475fff3663c

File: src/Model/PickUpPlace/PickUpPlaceIdToEntityTransformer.php [hunk 6]
```diff
@@ -2,30 +2,30 @@
...
     public function __construct(PickUpPlaceFacade $pickUpPlaceFacade)
     {
         $this->pickUpPlaceFacade = $pickUpPlaceFacade;
     }
 
     /**
-     * @var \Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlace
      * @param mixed $pickUpPlace
      * @return int|null
+     *@var \App\Model\PickUpPlace\PickUpPlace
```
Comment (pesektomas) [inline src/Model/PickUpPlace/PickUpPlaceIdToEntityTransformer.php]:
```suggestion
     * @var \App\Model\PickUpPlace\PickUpPlace
```

File: config/bundles.php [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    VasekPurchart\ConsoleErrorsBundle\ConsoleErrorsBundle::class => ['all' => true],
+    FOS\CKEditorBundle\FOSCKEditorBundle::class => ['all' => true],
+    Joschi127\DoctrineEntityOverrideBundle\Joschi127DoctrineEntityOverrideBundle::class => ['all' => true],
+    Shopsys\FrameworkBundle\ShopsysFrameworkBundle::class => ['all' => true],
+    Shopsys\ReadModelBundle\ShopsysReadModelBundle::class => ['all' => true],
+    Symfony\Bundle\WebProfilerBundle\WebProfilerBundle::class => ['dev' => true],
+    Symfony\Bundle\DebugBundle\DebugBundle::class => ['dev' => true],
+    Symfony\Bundle\WebServerBundle\WebServerBundle::class => ['dev' => true],
+    Symfony\WebpackEncoreBundle\WebpackEncoreBundle::class => ['all' => true],
+    Fp\JsFormValidatorBundle\FpJsFormValidatorBundle::class => ['all' => true],
```
Comment (pesektomas) [inline config/bundles.php]: I think so FpJsFormValidatorBundle have to be loaded before ShopsysFramework bundle, becouse ShopsysFramework bundle overrides some parameters.

File: config/packages/snc_redis.yaml [hunk 8]
```diff
@@ -11,36 +11,29 @@ snc_redis:
...
             options:
-                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:framework_annotations:'
+                prefix: '%env(REDIS_PREFIX)%%build-version%:cache:framework:annotations:'
         global:
             type: 'phpredis'
             alias: 'global'
             dsn: 'redis://%redis_host%'
             options:
                 prefix: '%env(REDIS_PREFIX)%'
-        twig:
```
Comment (pesektomas) [inline config/packages/snc_redis.yaml]: Why did you remove this configuration? I see it in project-base :thinking:

File: config/services.yaml [hunk 9]
```diff
@@ -1,85 +1,87 @@
...
-    Shopsys\ShopBundle\Model\Country\CountryRepository: ~
+    Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserDataFactoryInterface: '@App\Model\Customer\User\CustomerUserDataFactory'
 
-    Shopsys\FrameworkBundle\Model\Country\CountryRepository: '@App\Model\Country\CountryRepository'
+    Shopsys\FrameworkBundle\Model\Order\OrderDataFactoryInterface: '@App\Model\Order\OrderDataFactory'
 
-    Shopsys\FrameworkBundle\Model\Transport\TransportFactoryInterface: '@App\Model\Transport\TransportFactory'
+    Shopsys\FrameworkBundle\Model\Order\Item\OrderItemDataFactoryInterface: '@App\Model\Order\Item\OrderItemDataFactory'
+
+    App\Model\Order\OrderDataMapper: ~
```
Comment (pesektomas) [inline config/services.yaml:54]: Is it necessary? Mapper is included in autowiring `resource: '../src/**/*{Calculation,...,Mapper,Parser,...`

File: config/services.yaml [hunk 10]
```diff
@@ -114,21 +131,17 @@ services:
...
+    App\Model\Order\FrontOrderData: ~
 
     Shopsys\FrameworkBundle\Model\Order\FrontOrderData: '@App\Model\Order\FrontOrderData'
 
-    Shopsys\ShopBundle\Model\Order\OrderDataMapper: ~
-
-    Shopsys\FrameworkBundle\Model\Order\OrderDataMapper: '@App\Model\Order\OrderDataMapper'
-
-    Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceFacade: ~
+    App\Model\PickUpPlace\PickUpPlaceFacade: ~
```
Comment (pesektomas) [inline config/services.yaml]: Is it necessary? Facade is included in autowiring

File: config/services.yaml [hunk 11]
```diff
@@ -114,21 +131,17 @@ services:
...
 
-    Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceIdToEntityTransformer: ~
+    App\Model\PickUpPlace\PickUpPlaceIdToEntityTransformer: ~
 
-    Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceLoader: ~
+    App\Model\PickUpPlace\PickUpPlaceLoader: ~
 
-    Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceRepository: ~
+    App\Model\PickUpPlace\PickUpPlaceRepository: ~
```
Comment (pesektomas) [inline config/services.yaml]: Is it necessary? Repository is included in autowiring

File: config/services.yaml [hunk 12]
```diff
@@ -114,21 +131,17 @@ services:
...
-    Shopsys\FrameworkBundle\Model\Order\OrderDataMapper: '@App\Model\Order\OrderDataMapper'
-
-    Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceFacade: ~
+    App\Model\PickUpPlace\PickUpPlaceFacade: ~
 
-    Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceIdToEntityTransformer: ~
+    App\Model\PickUpPlace\PickUpPlaceIdToEntityTransformer: ~
 
-    Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceLoader: ~
+    App\Model\PickUpPlace\PickUpPlaceLoader: ~
```
Comment (pesektomas) [inline config/services.yaml]: Is it necessary? Loader is included in autowiring

File: config/services.yaml [hunk 13]
```diff
@@ -114,21 +131,17 @@ services:
...
 
-    Shopsys\ShopBundle\Model\Order\OrderDataMapper: ~
-
-    Shopsys\FrameworkBundle\Model\Order\OrderDataMapper: '@App\Model\Order\OrderDataMapper'
-
-    Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceFacade: ~
+    App\Model\PickUpPlace\PickUpPlaceFacade: ~
 
-    Shopsys\ShopBundle\Model\PickUpPlace\PickUpPlaceIdToEntityTransformer: ~
+    App\Model\PickUpPlace\PickUpPlaceIdToEntityTransformer: ~
```
Comment (pesektomas) [inline config/services.yaml]: Is it necessary? Transformer is included in autowiring

File: config/services.yaml [hunk 14]
```diff
@@ -139,29 +152,29 @@ services:
...
         arguments:
             - "%shopsys.performance_data.product.total_count%"
 
-    Shopsys\ShopBundle\DataFixtures\Performance\UserDataFixture:
+    App\DataFixtures\Performance\UserDataFixture:
         arguments:
             - "%shopsys.performance_data.user.count_per_domain%"
 
-    Shopsys\ShopBundle\DataFixtures\Demo\ImageDataFixture:
+    App\DataFixtures\Demo\ImageDataFixture:
```
Comment (pesektomas) [inline config/services.yaml:153]: Are not datafixtures declrating in `config/services/data_fixtures.yml` ?

File: composer.json [hunk 15]
```diff
@@ -49,93 +43,132 @@
         "doctrine/doctrine-bundle": "^1.8.1",
         "doctrine/doctrine-fixtures-bundle": "^3.0.2",
         "doctrine/doctrine-migrations-bundle": "^1.3.0",
-        "fp/jsformvalidator-bundle": "^1.5.1",
+        "doctrine/persistence": "~1.2.0",
+        "fp/jsformvalidator-bundle": "dev-master",
```
Comment (pesektomas) [inline composer.json]: current version of this bundle is 1.6.1

File: composer.json [hunk 16]
```diff
@@ -49,93 +43,132 @@
...
-            "Shopsys\\Environment::checkEnvironment",
-            "Shopsys\\FrameworkBundle\\Command\\ComposerScriptHandler::postUpdate"
-        ]
+            "App\\Environment::checkEnvironment",
+            "@auto-scripts"
+        ],
+        "auto-scripts": {
+            "shopsys:domains-urls:configure": "symfony-cmd",
+            "ckeditor:install --clear=skip --release=full --tag=4.5.11": "symfony-cmd",
+            "cache:clear --no-warmup": "symfony-cmd",
```
Comment (pesektomas) [inline composer.json]: You can use this `"php phing clean": "script",` and move it to first place :smile:

File: phpunit.xml [hunk 17]
```diff
@@ -1,34 +1,37 @@
...
+        backupStaticAttributes="false"
+        bootstrap="config/bootstrap.php"
+        colors="true"
+        convertErrorsToExceptions="true"
+        convertNoticesToExceptions="true"
+        convertWarningsToExceptions="true"
+        processIsolation="false"
+        stopOnFailure="false"
+        failOnWarning="true"
+        beStrictAboutTestsThatDoNotTestAnything="false"
```
Comment (pesektomas) [inline phpunit.xml]: I think so this indentation is broken

File: easy-coding-standard.yml [hunk 18]
```diff
@@ -17,34 +21,43 @@ parameters:
...
+            - '*/tests/App/*'
+
+        PHP_CodeSniffer\Standards\PSR2\Sniffs\Methods\MethodDeclarationSniff.Underscore:
+            - '*/tests/App/Test/Codeception/Helper/CloseNewlyOpenedWindowsHelper.php'
+            - '*/tests/App/Test/Codeception/Helper/DatabaseHelper.php'
+            - '*/tests/App/Test/Codeception/Helper/DomainHelper.php'
+            - '*/tests/App/Test/Codeception/Helper/LocalizationHelper.php'
+            - '*/tests/App/Test/Codeception/Helper/NumberFormatHelper.php'
+            - '*/tests/App/Test/Codeception/Helper/SymfonyHelper.php'
+            - '*/tests/App/Test/Codeception/Module/Db.php'
```
Comment (pesektomas) [inline easy-coding-standard.yml:64]: This is perfect commit :smile: Commit message is `Fix some standarts` and you updated ignore paths :smile:

File: Gruntfile.js [hunk 19]
```diff
@@ -0,0 +1,431 @@
+module.exports = function(grunt) {
```
Comment (pesektomas) [inline Gruntfile.js]: You can remove thi file

**Review by pesektomas — CHANGES_REQUESTED** [review]

File: config/bundles.php (hunk 7 above)
Comment (pesektomas) [reply config/bundles.php]: This is resolved in some commit, sorry

File: config/packages/snc_redis.yaml (hunk 8 above)
Comment (pesektomas) [reply config/packages/snc_redis.yaml]: This block was get back in some commit ....

File: config/services.yaml (hunk 10 above)
Comment (TomasLudvik) [reply config/services.yaml]: removed

File: config/services.yaml (hunk 11 above)
Comment (TomasLudvik) [reply config/services.yaml]: removed

File: config/services.yaml (hunk 12 above)
Comment (TomasLudvik) [reply config/services.yaml]: removed

File: src/Model/PickUpPlace/PickUpPlaceIdToEntityTransformer.php (hunk 6 above)
Comment (TomasLudvik) [reply src/Model/PickUpPlace/PickUpPlaceIdToEntityTransformer.php]: fixed

File: config/services.yaml (hunk 13 above)
Comment (TomasLudvik) [reply config/services.yaml]: removed

File: config/services.yaml (hunk 14 above)
Comment (TomasLudvik) [reply config/services.yaml:153]: moved

File: composer.json (hunk 16 above)
Comment (TomasLudvik) [reply composer.json]: done

File: phpunit.xml (hunk 17 above)
Comment (TomasLudvik) [reply phpunit.xml]: fixed

File: easy-coding-standard.yml (hunk 18 above)
Comment (TomasLudvik) [reply easy-coding-standard.yml:64]: That fixed the problem! :)

File: config/services.yaml (hunk 14 above)
Comment (TomasLudvik) [reply config/services.yaml:153]: reverted - caused some weird problem

Comment (pesektomas) [reply config/services.yaml:153]: ok

File: easy-coding-standard.yml (hunk 18 above)
Comment (pesektomas) [reply easy-coding-standard.yml:64]: It can fixed all problems :)

**Review by pesektomas — APPROVED** [review]

**Review by sslt — APPROVED** [review]
